core/src/main/clojure/xtdb/buffer_pool.clj

```diff
@@ -42,21 +48,45 @@
 
 (def ^java.nio.file.Path storage-root (util/->path version))
 
-(defn- retain [^ArrowBuf buf] (.retain (.getReferenceManager buf)) buf)
-
 (defn ->buffer-pool-child-allocator [^BufferAllocator allocator, ^MeterRegistry metrics-registry]
   (let [child-allocator (.newChildAllocator allocator "buffer-pool" 0 Long/MAX_VALUE)]
     (when metrics-registry
       (metrics/add-allocator-gauge metrics-registry "buffer-pool.allocator.allocated_memory" child-allocator))
     child-allocator))
 
+(defn copy-bb ^ByteBuffer [^ByteBuffer bb]
+  (let [copy (ByteBuffer/allocate (.remaining bb))]
```

Comment:
ensure this doesn't mutate the position of the original bb - if so, you might want to `duplicate` it before you use it.

Reply:
When I apply the `ByteBuffer` to `ByteArray` chang above, this will go away anyway.